lib/fs/basicfs.go

```diff
@@ -87,6 +88,7 @@ func newBasicFilesystem(root string, opts ...Option) *BasicFilesystem {
 		root:    root,
 		options: opts,
 	}
+	fs.OSDataGetter = NewOSDataGetter(fs)
```

Comment:
This is somewhat grim, could we not have platform specific implementations for this, simillar like we have for notify etc?